core/src/main/java/com/segment/analytics/kotlin/core/HTTPClient.kt

```diff
@@ -78,11 +78,12 @@ abstract class Connection(
     }
 }
 
-fun safeGetInputStream(connection: HttpURLConnection): InputStream {
+fun safeGetInputStream(connection: HttpURLConnection): InputStream? {
     return try {
         connection.inputStream
     } catch (ignored: IOException) {
-        connection.errorStream
```

Comment:
im a bit confused on this change

Reply:
ah. yeah. this one is a weird one. if you check the source of `connection.errorStream`, it's actually returning null.
```java
    public InputStream getErrorStream() {
        return null;
    }
```
so we need to change the return type to `InputStream?`, or it will throws an exception and crashes the sdk. also since `connection.stream` always returning null, we can just return null in our code as well. but returning `connection.errorStream` also works and has better readability

Reply:
lets keep it as `connection.errorStream`. The default implementation might return `null` but i believe that method can be overridden by other implementations which may not return `null`.

Reply:
yeah. good point. that can totally happen. making the change now